hydra-plutus/src/Hydra/Contract/Head.hs

```diff
@@ -380,32 +400,6 @@ checkFanout utxoHash contestationDeadline numberOfFanoutOutputs ScriptContext{sc
 (&) = flip ($)
 {-# INLINEABLE (&) #-}
 
--- | Check that the output datum of this script corresponds to an expected
--- value. Takes care of resolving datum hashes and inline datums.
-checkHeadOutputDatum :: ToData a => ScriptContext -> a -> Bool
-checkHeadOutputDatum ctx d =
```

Comment:
I still find this check useful to maintain, possibly as the very last validation step for the script.

I know we have made some simplifications that makes this check potentially redundant, but I still see value to keep it at the very end.

wdyt?

Reply:
We pay for ever bit of code included on-chain, we do not want to keep redundant checks.

Reply:
@ffakenz do you see something that we used to test before that wouldn't be checked by this version of the code? If so, would you share an example? That could be helpful to double-check our mutation tests.

Reply:
I do not see a counter-example, but this check looks like a catch-all to me, that is the reason why I suggest keeping it.

Reply:
A conversation later in this P.R. mentions that, indeed, we might have a problem with headId and parties being corrupted. I'll resove the conversation here and move it to this [other one](https://github.com/input-output-hk/hydra/pull/677#discussion_r1091213101)